gencode.c

```diff
@@ -1747,6 +1747,13 @@ init_linktype(compiler_state_t *cstate, pcap_t *p)
 		cstate->off_nl_nosnap = 3;	/* 802.3+802.2 */
 		break;
 
+	case DLT_DSA_TAG_GSW1XX:
+            cstate->off_linktype.constant_part = 20;
+		cstate->off_linkpl.constant_part = 22;	/* Ethernet header length + Special Tag Length */
+		cstate->off_nl = 0;		/* Ethernet II */
+		cstate->off_nl_nosnap = 3;	/* 802.3+802.2 */
+		break;
+
```

Comment:
Packet filtering is a bit more complicated than that (see #1195), but it does not have to be implemented right at the start. I suggest to attempt filtering as one of the next steps.